samples/subsys/ipc/openamp/remote_echo/README.rst

```diff
@@ -0,0 +1,142 @@
+.. _openAMP_remote_echo_sample:
+
+OpenAMP Remote Echo Sample Application
+######################################
+
+Overview
+********
+
+This sample application demonstrates how to integrate coding and building of
+OpenAMP with Zephyr. Currently this integration is specific to the i.MX7 and
+i.MX6 SoCs and the Master side (Cortex-A) should be running Linux with the i.MX
+RPMsg and mailbox driver.
+
+
+Building the application
+*************************
+
+.. zephyr-app-commands::
+    :zephyr-app: samples/subsys/ipc/openamp/remote_echo
+    :board: colibri_imx7d_m4 warp7_m4 udoo_neo_full_m4
+    :goals: build flash
+
+
+On the Linux side (Master) make sure you have the following Linux kernel config
+options enabled:
+
+.. code-block:: none
+
+    CONFIG_HAVE_IMX_MU=y
+    CONFIG_HAVE_IMX_RPMSG=y
+    CONFIG_RPMSG=y
+    CONFIG_RPMSG_VIRTIO=y
+    CONFIG_IMX_RPMSG_TTY=m
```

Comment:
Is there any reason to enable IMX_RPMST_TTY driver?
Just check the .c file, it looks like it implements the echo, from meta-openamp, there is userspace demo which uses the kernel rpmsg_char driver, as @arnop2 mentioned, there is also kernel sample driver.
Here is the OpenAMP apps for the kernel sample driver: https://github.com/OpenAMP/open-amp/blob/master/apps/examples/rpmsg_sample_echo/rpmsg-sample-ping.c

It looks like only the virtio operation implementation is different, maybe you can reuse that application.